Src/UnitTests/CoravelUnitTests/Scheduling/IntervalTests/SchedulerCronTests.cs

```diff
@@ -13,128 +14,154 @@ public class SchedulerCronTests
+        [InlineData("* * * * 0", "8/18/2018 10:59 pm", false, false)]
+        [InlineData("* * * * 0", "8/19/2018 10:59 pm", true, false)]
+        [InlineData("* * * * 0,6", "8/12/2018 10:59 pm", true, false)]
+        [InlineData("* * * * 0,6", "8/13/2018 10:59 pm", false, false)]
+        [InlineData("* * * * 0,6", "8/18/2018 10:59 pm", true, false)]
+        [InlineData("* * * * 2-4", "8/14/2018 10:59 pm", true, false)]
+        [InlineData("* * * * 2-4", "8/15/2018 10:59 pm", true, false)]
+        [InlineData("* * * * 2-4", "8/16/2018 10:59 pm", true, false)]
+        [InlineData("* * * * 2-4", "8/17/2018 10:59 pm", false, false)]
         // Mixed
-        [InlineData("00 01 * * *", "8/14/2018 1:00 am", true)]
-        [InlineData("00 01 * * *", "8/14/2018 1:01 am", false)]
-        [InlineData("05 01 * * *", "8/14/2018 1:05 am", true)]
-        [InlineData("05 01 * * *", "8/14/2018 1:06 am", false)]
-        [InlineData("05 01 * * *", "8/14/2018 1:05 pm", false)]
-        [InlineData("00 00 01 * *", "8/1/2018 12:00 am", true)]
-        [InlineData("00 00 01 * *", "8/2/2018 12:00 am", false)]
-        [InlineData("00 00 01 02 *", "8/1/2018 12:00 am", false)]
-        [InlineData("00 00 01 02 *", "2/1/2018 12:00 am", true)]
-        [InlineData("00 00 * * 0", "8/19/2018 12:00 am", true)]
-        [InlineData("00 00 * * 0", "8/20/2018 12:00 am", false)]
-        [InlineData("00 00 */2 * *", "8/2/2018 12:00 am", true)]
-        [InlineData("00 00 */2 * *", "8/12/2018 12:00 am", true)]
-        [InlineData("00 00 */2 * *", "8/3/2018 12:00 am", false)]
-        [InlineData("00 00 */2 * *", "8/13/2018 12:00 am", false)]
-        public async Task ScheduledEventCron(string cronExpression, string dateString, bool shouldRun)
+        [InlineData("00 01 * * *", "8/14/2018 1:00 am", true, false)]
+        [InlineData("00 01 * * *", "8/14/2018 1:01 am", false, false)]
+        [InlineData("05 01 * * *", "8/14/2018 1:05 am", true, false)]
+        [InlineData("05 01 * * *", "8/14/2018 1:06 am", false, false)]
+        [InlineData("05 01 * * *", "8/14/2018 1:05 pm", false, false)]
+        [InlineData("00 00 01 * *", "8/1/2018 12:00 am", true, false)]
+        [InlineData("00 00 01 * *", "8/2/2018 12:00 am", false, false)]
+        [InlineData("00 00 01 02 *", "8/1/2018 12:00 am", false, false)]
+        [InlineData("00 00 01 02 *", "2/1/2018 12:00 am", true, false)]
+        [InlineData("00 00 * * 0", "8/19/2018 12:00 am", true, false)]
+        [InlineData("00 00 * * 0", "8/20/2018 12:00 am", false, false)]
+        [InlineData("00 00 */2 * *", "8/2/2018 12:00 am", true, false)]
+        [InlineData("00 00 */2 * *", "8/12/2018 12:00 am", true, false)]
+        [InlineData("00 00 */2 * *", "8/3/2018 12:00 am", false, false)]
+        [InlineData("00 00 */2 * *", "8/13/2018 12:00 am", false, false)]
+        // Mixed, TimeZoneAware
+        [InlineData("00 01 * * *", "8/14/2018 12:00 am", true, true)]
```

Comment:
I think we should test a few actual scenarios where DST is involved to make sure the cases aren't broken in the future.

Reply:
well the problem here is that it's really really hard to make test cases with a real time zone, that won't break in the summer or in the winter. 
But I try to figure something out. I guess it might also make sense to have a test file only for the `CronWithTimeZone`.

Reply:
well the problem it's basically impossible to test for dst/non-dst.
what I can do is use a real timezone like the Central European Standard Time and use either the `DST` string or the normal one, unfortunatly it's impossible to test that without changing the computer clock.

it would also possible to add `AdjustmentRules` to `CreateCustomTimeZone` unfortunatly that is extremly hard.
So the only thing would be to create tests that are different in winter or summer or I can create the fake time zone with different `TimeSpan.FromHours` which is basically nothing else than a "dst" switch